src/main/js/utils/Schedule.js

```diff
@@ -11,13 +11,33 @@ class Schedule extends Set {
       super(data);
       this.gap = data.gap;
       this.credits = data.credits;
+      this.gpa = 0.00;
     } else {
       super();
       this.gap = data;
       this.credits = 0;
+      this.gpa = 0.00;
     }
   }
 
+  /**
+   * Function to calculate the estimated gpa of a schedule
+   */
+  calculateGPA() {
```

Comment:
We can keep this function and pass in the map of CourseGPAs I was talking about. Also I believe you need to reset the gpa field in the clear method.

Reply:
Moved to another class as an exportable function taking in a schedule and gradeMap